pythran/config.py

```diff
@@ -21,8 +24,48 @@ def init_cfg(sys_file, platform_file, user_file):
         cfg.readfp(open(required))
     cfg.read([user_config_path])
 
+    for key in ('CC', 'CXX'):
+        value = cfg.get('compiler', key)
+        if value:
+            os.environ[key] = value
+
+    for obsolete_section in ('user', 'sys'):
+        if cfg.has_section(obsolete_section):
+            logger.warn("Your pythranrc has an obsolete `{}' section".format(
+                obsolete_section))
+
     return cfg
 
+
+def make_extension(**extra):
+    def parse_define(define):
+        index = define.find('=')
+        if index < 0:
+            return (define, None)
+        else:
+            return define[:index], define[index + 1:]
+    extension = {
+        "define_macros": cfg.get('compiler', 'defines').split(),
+        "undef_macros": cfg.get('compiler', 'undefs').split(),
+        "include_dirs": cfg.get('compiler', 'include_dirs').split(),
+        "library_dirs": cfg.get('compiler', 'library_dirs').split(),
+        "libraries": cfg.get('compiler', 'libs').split(),
+        "extra_compile_args": cfg.get('compiler', 'cflags').split(),
+        "extra_link_args": cfg.get('compiler', 'ldflags').split(),
+    }
+
+    extension['define_macros'].append('ENABLE_PYTHON_MODULE')
+
+    here = os.path.dirname(os.path.dirname(__file__))
+    extension["include_dirs"].append(here + '/pythran')
```

Comment:
I admit it is pedantic. It is just because when I see code like this, I think "Programmer was lazy so I will fix it and use os.path.join" While if you use posixpath, I know it is intensional and that it should not be changed.